src/arrayMethodSort.js

```diff
@@ -4,8 +4,31 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    const defaultCompareFun = (a, b) => {
+      const strA = String(a);
+      const strB = String(b);
+
+      return strA < strB ? -1 : strA > strB ? 1 : 0;
+    };
+
+    const compare =
+      typeof compareFunction === 'function'
+        ? compareFunction
+        : defaultCompareFun;
+
+    for (let i = 0; i < this.length - 1; i++) {
+      for (let j = 0; j < this.length - 1 - i; j++) {
+        if (compare(this[j], this[j + 1]) > 0) {
+          const temp = this[j];
+
+          this[j] = this[j + 1];
+          this[j + 1] = temp;
+        }
+      }
+    }
+
+    return this;
   };
```

Comment:
The method sort2 is being assigned to [].__proto__, which is generally not recommended. Modifying an object's prototype, especially Array.prototype, can lead to unpredictable behavior in code that depends on the original methods. It's better to create a standalone function or use a different approach to extend the Array prototype without affecting the built-in methods.